### PR TITLE
[MM-29388][MM-29387]Campaign/applytheme sidebar bg

### PR DESCRIPTION
**Pull request by deanwhillier**
#### Summary
ApplyTheme Campaign: This PR completes the merge of the theme.sidebarHeaderBg commits from the campaign branch.

Dev approval for the changes was completed on the original PR(s), so we are just looking for rubber stamp approvals by dev's to satisfy merge requirements.

As QA was skipped on PR merges into campaign branches, this PR will need proper QA before merging. See below for details on what to QA for this PR.

#### Ticket Link

* https://github.com/mattermost/mattermost-server/issues/15826
* https://github.com/mattermost/mattermost-server/issues/15827

#### Related Pull Requests
See above tickets for related PRs

### CODE REVIEW

File: sass/responsive/_mobile.scss [hunk 1]
```diff
@@ -16,8 +16,18 @@
                 border: none;
             }
         }
+     .modal {
+         .settings-modal {
+             &:not(.settings-modal--tables) {
+                 &:not(.display--content) {
+                     .modal-content {
+                         background: var(--sidebar-bg);
```
Comment (deanwhillier) [inline sass/responsive/_mobile.scss:24]: @jgilliam17, this seems to affect the background of the account settings modal in screen widths less than 768px. It actually manifests as what visually appears to be the border colour though.

File: sass/responsive/_mobile.scss [hunk 2]
```diff
@@ -16,8 +16,18 @@
                 border: none;
             }
         }
+     .modal {
```
Comment (devinbinnie) [inline sass/responsive/_mobile.scss:19]: Nit: indenting looks off here, should be 8 spaces instead of 5

File: sass/responsive/_mobile.scss [hunk 3]
```diff
@@ -16,8 +16,18 @@
...
+         .settings-modal {
+             &:not(.settings-modal--tables) {
```
Comment (devinbinnie) [inline sass/responsive/_mobile.scss:21]: Typo
```suggestion
             &:not(.settings-modal--tabless) {
```

File: utils/utils.jsx [hunk 4]
```diff
@@ -502,8 +502,6 @@ export function applyTheme(theme) {
     if (theme.sidebarBg) {
         changeCss('.app__body .sidebar--left .sidebar__switcher, .sidebar--left, .app__body .modal .settings-modal .settings-table .settings-links, .app__body .sidebar--menu', 'background:' + theme.sidebarBg);
         changeCss('body.app__body', 'scrollbar-face-color:' + theme.sidebarBg);
-        changeCss('@media(max-width: 768px){.app__body .modal .settings-modal:not(.settings-modal--tabless):not(.display--content) .modal-content', 'background:' + theme.sidebarBg);
-        changeCss('.app__body .modal-tabs .nav-tabs > li.active', `border-bottom-color:${theme.sidebarBg}`);
```
Comment (devinbinnie) [inline utils/utils.jsx:506]: Is there not supposed to be any CSS to replace this?

**Review by devinbinnie — CHANGES_REQUESTED** [review]

File: utils/utils.jsx (hunk 4 above)
Comment (deanwhillier) [reply utils/utils.jsx:506]: [Original PR](https://github.com/mattermost/mattermost-webapp/pull/6698) determined that line was no longer in use.

File: sass/responsive/_mobile.scss (hunk 3 above)
Comment (deanwhillier) [reply sass/responsive/_mobile.scss:21]: Good catch!

**Review by devinbinnie — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @deanwhillier 
Tested, looks good to merge.
- Verified background color of the account settings modal in screen smaller screen widths.

File: sass/responsive/_mobile.scss [hunk 5]
```diff
@@ -16,8 +16,18 @@
                 border: none;
             }
         }
+        .modal {
+            .settings-modal {
+                &:not(.settings-modal--tabless) {
+                    &:not(.display--content) {
+                        .modal-content {
```
Comment (hmhealey) [inline sass/responsive/_mobile.scss:23]: Hopefully we can make this a lot cleaner and not so deeply nested in the future.

**Review by hmhealey — APPROVED** [review]

**Pull request merged**